File: Array/Reversal Algorithm/reversal.py
def reverseArray(arr, start, end):
    while (start < end):
        temp = arr[start]
        arr[start] = arr[end]
        arr[end] = temp
        start += 1
        end = end-1
  
def leftRotate(arr, d):
  
    if d == 0:
        return
    n = len(arr)
    d = d % n
    reverseArray(arr, 0, d-1)
    reverseArray(arr, d, n-1)
    reverseArray(arr, 0, n-1)
  
  
def printArray(arr):
    for i in range(0, len(arr)):
        print (arr[i],end=' ')

arr = [1, 2, 3, 4, 5, 6, 7]
n = len(arr)
d = 2
  
leftRotate(arr, d)
printArray(arr)

File: Array/Reversal Algorithm/Reversal.java
public class Reversal_Algorithm {

    public static void main(String[] args) {
        int[] arr = {1,2 ,3 ,4,5 ,6 ,7};
        System.out.println(Arrays.toString(arr));
        reverse_array(arr);
        System.out.println(Arrays.toString(arr));
    }

    static void reverse_array(int[] arr) {
        int s = 0;
        int e = arr.length - 1;
        while (s < e) {

            int t = arr[s];
            arr[s] = arr[e];
            arr[e] = t;

            s++;
            e--;
        }
    }
}
